usr/lib/zypp/plugins/urlresolver/susecloud

```diff
@@ -42,10 +42,10 @@ class SUSECloudPlugin(Plugin):
         if zypper_pid != prev_zypper_pid:
             verify_credentials = True
         # Note this logic breaks when FATE#320882/PM-1251 gets implemented
-        if verify_credentials and not utils.credentials_files_are_equal():
+        if verify_credentials and not utils.credentials_files_are_equal(headers.get('credentials')):
```

Comment:
Create a temporary `repo_credentials = headers.get('credentials')` and then also re-use this cached value on lines 67 and 70 and remove line 65.

i.e.:
`repo_url += '?credentials=' + repo_cerdentials`
'/etc/zypp/credentials.d/%s' % repo_credentials

Reply:
done